lib/handler/compress/gzip.c

```diff
@@ -74,16 +81,16 @@ static size_t compress_chunk(struct st_gzip_context_t *self, const void *src, si
         }
         self->zs.next_out = (void *)(self->bufs.entries[bufindex].base + self->bufs.entries[bufindex].len);
         self->zs.avail_out = (unsigned)(BUF_SIZE - self->bufs.entries[bufindex].len);
-        ret = deflate(&self->zs, flush);
-        assert(ret == Z_OK || ret == Z_STREAM_END);
+        ret = proc(&self->zs, flush);
+        assert(ret == Z_OK || ret == Z_STREAM_END || ret == Z_BUF_ERROR);
```

Comment:
Thank you for looking into the issue.

So to clarify, it seems that `inflate()` returns `Z_BUF_ERROR` if `flush` is set to `Z_FINISH` at the middle of the compressed data (quote: will return Z_BUF_ERROR if it has not reached the end of the stream).

If that is the case, I have no objections to the change. OTOH, I would appreciate if you could add a in-line comment to the source code explaining why such change is necessary.

Thank you in advance.